tests/integration/executors/test_celery_executor.py

```diff
@@ -107,6 +110,26 @@ def teardown_method(self) -> None:
         db.clear_db_runs()
         db.clear_db_jobs()
 
+    def test_change_state_back_compat(self):
+        # This represents the old implementation that an Airflow package may have
+        def _change_state(self, key: TaskInstanceKey, state: TaskInstanceState, info=None) -> None:
+            pass
+
+        # Replace change_state function on base executor with the old version to force the backcompat edge
+        # case we're looking for
+        base_executor.BaseExecutor.change_state = _change_state
```

Comment:
I couldn't get this working with a mock. If I tried to create a BaseExecutor mock, it would in turn cause the CeleryExecutor to become a mock, I'm assuming because it calls the super().__init__() of base executor.

So I went with monkey patching this method, which should be pretty safe (now) since the back compat code is working :)

Reply:
How about just make `change_state` on the celery executor a mock? You can do that with `mock.patch.object`

Reply:
But that's the code under test, if I mock that I'm no longer testing my actual changes

Reply:
You can use `side_effect` to still call the actual implementation while recording the call count. I don’t have a preference though, just adding possibilities if you’re looking for alternatives.